lib/plugins/aws/deploy/compile/functions/index.js

```diff
@@ -104,7 +104,8 @@ class AwsCompileFunctions {
       const functionObject = this.serverless.service.getFunction(functionLogicalName);
 
       newFunction.Properties.Code
-        .S3Key = this.serverless.service.package.artifact.split(path.sep).pop();
+        .S3Key = `${this.serverless.service.package.artifactDirectoryName
+        }/${this.serverless.service.package.artifact.split(path.sep).pop()}`;
```

Comment:
We should not use the filename of the artefact, but create our own filename. Otherwise files might not be correctly renamed and CF won't detect an update. This also means we have more control over it